src/User32.Desktop/User32+SafeWindowStationHandle.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SafeWindowStationHandle"/> nested type.
+    /// </content>
+    public static partial class User32
+    {
+        /// <summary>
+        /// Represents a Desktop handle that can be closed with <see cref="CloseWindowStation(IntPtr)"/>.
+        /// </summary>
+        public class SafeWindowStationHandle : SafeHandle
+        {
+            /// <summary>
+            /// A handle that may be used in place of <see cref="IntPtr.Zero"/>.
+            /// </summary>
+            public static readonly SafeWindowStationHandle Null = new SafeWindowStationHandle(IntPtr.Zero, false);
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SafeWindowStationHandle"/> class.
+            /// </summary>
+            public SafeWindowStationHandle()
+                : base(IntPtr.Zero, true)
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SafeWindowStationHandle"/> class.
+            /// </summary>
+            /// <param name="preexistingHandle">An object that represents the pre-existing handle to use.</param>
+            /// <param name="ownsHandle">
+            ///     <see langword="true" /> to have the native handle released when this safe handle is disposed or finalized;
+            ///     <see langword="false" /> otherwise.
+            /// </param>
+            public SafeWindowStationHandle(IntPtr preexistingHandle, bool ownsHandle = true)
+                : base(IntPtr.Zero, ownsHandle)
+            {
+                this.SetHandle(preexistingHandle);
+            }
+
+            public override bool IsInvalid => this.handle == IntPtr.Zero;
+
+            protected override bool ReleaseHandle()
+            {
+                if (this.IsInvalid)
+                {
+                    return false;
+                }
+
+                bool closed = CloseWindowStation(this.handle);
```

Comment:
I'm not sure that there should be anything but the call to CloseWindowStation there, it's what's in the sample at the top of the SafeHandle source code : http://referencesource.microsoft.com/#mscorlib/system/runtime/interopservices/safehandle.cs,85 and as it can run in very special cases (In a CER, on the finalizer thread)

For the specific of your 2 checks they aren't useful at least if I read coreclr code well (And I believe all microsoft implementations do the same, didn't check Mono) :
* Double dispose it's not possible, the comment here describe why :  https://github.com/dotnet/coreclr/blob/775003a4c72f0acc37eab84628fcef541533ba4e/src/vm/safehandle.cpp#L81
* For checking IsInvalid, it's already done : https://github.com/dotnet/coreclr/blob/775003a4c72f0acc37eab84628fcef541533ba4e/src/vm/safehandle.cpp#L199